Final (0.2.0) cutting workflow (#153)
Qiskit-Extensions/circuit-knitting-toolbox: docs/circuit_cutting/explanation/index.rst

> ###################################################

Author: @garrison I think the QPD explanatory can/should go in here, rather than having an extra explanatory. We can just have a QPD section after the overview I think

###################################################
Explanatory material for the circuit cutting module
###################################################

Overview of circuit cutting
---------------------------

> Circuit cutting is a technique to increase the size of circuits we can run on quantum hardware at the cost of an additional sampling overhead. A larger quantum circuit can be decomposed by cutting its gates and wires into smaller circuits, which can be executed within the constraints of available quantum hardware. The results of these smaller circuits are combined to reconstruct the outcome of the original problem. Circuit cutting can also be used to engineer gates between distant qubits which would otherwise require a large swap overhead.

Reviewer: Just a suggestion
```suggestion
Circuit cutting is a technique to increase the size of circuits we can run on quantum hardware at the cost of an additional sampling overhead. A larger quantum circuit can be decomposed by cutting its gates and/or wires, resulting in smaller circuits which can be executed within the constraints of available quantum hardware. The results of these smaller circuits are combined to reconstruct the outcome of the original problem. Circuit cutting can also be used to engineer gates between distant qubits which would otherwise require a large swap overhead.
```

> For more detailed information on this technique, refer to the paper, Error mitigation for short-depth quantum circuits [`4 <https://journals.aps.org/prl/abstract/10.1103/PhysRevLett.119.180509>`_].

Reviewer: I like to use arXiv links for these things, because the arXiv paper is available without a subscription, and it always links to the real paper.  An alternative could be to include both links.  Same comment where the reference is listed at the end of this document.

https://arxiv.org/abs/1612.02058

Key terms
-----------------

> * subcircuits: The set of circuits resulting from cutting gates in a `QuantumCircuit` and then separating the disconnected qubit subsets into smaller circuits. These circuits contain `SingleQubitQPDGate`\ s and will be used to instantiate each unique subexperiment.

Reviewer: need to either put double backticks around `QuantumCircuit` or put `:class:` immediately before the first backtick.  Same for the other classes that follow.

* subexperiments: A term used to describe the unique circuit samples associated with a subcircuit. These circuits have had their `BaseQPDGate`\ s decomposed into local Qiskit gates and measurements. Subexperiments are the circuits sent to the backend for execution.

* decompose: We try to honor the Qiskit notion of "decompose" in the documentation and API, which loosely means transforming a gate into a less-abstracted representation. *Ocassionally*, we may use the term "decompose" to refer to the act of inserting `BaseQPDGate` instances into quantum circuits as "decomposing" a gate or wire; however, we try to use terms like "partition" and "cut" when referring to this to avoid ambiguity with Qiskit language.

Current limitations
-------------------

> * QPD-based wire cutting available no sooner than CKT v0.3.0. The `cutqc <https://qiskit-extensions.github.io/circuit-knitting-toolbox/circuit_cutting/cutqc/index.htmlpackage>`_ package may be used for wire cutting in the meantime.

Reviewer: ```suggestion
* QPD-based wire cutting will be available no sooner than CKT v0.3.0. The `cutqc <https://qiskit-extensions.github.io/circuit-knitting-toolbox/circuit_cutting/cutqc/index.htmlpackage>`_ package may be used for wire cutting in the meantime.
```

> * `PauliList` is the only supported observable format until no sooner than CKT v.0.3.0

Reviewer: ```suggestion
* ``PauliList`` is the only supported observable format until no sooner than CKT v0.3.0.
```
double backticks since this is rst, plus a small tweak

There are two types of cuts: gate cuts and wire cuts.  Gate cuts, also known as "space-like" cuts, exist when the cut goes through a gate operating on two (or more) qubits.  Wire cuts, also known as "time-like" cuts, are direct cuts through a qubit wire, essentially a single-qubit identity gate that has been cut into two pieces.  A wire cut is simulated by introducing a new qubit into the circuit and moving remaining operations after the cut identity gate to the new qubit.

There are three settings to consider for circuit cutting.  The first is where only local operations (LO) [i.e., local _quantum_ operations] are available.  The other settings introduce classical communication between the circuit executions, which is known in the quantum information literature as LOCC, for `local operations and classical communication <https://en.wikipedia.org/wiki/LOCC>`__.  The LOCC can be either near-time, one-directional communication between the circuit executions (the second setting), or real-time, bi-directional communication (the third setting).

> As mentioned above, the cost of any simulation based on quasiprobability distribution is an exponential sampling overhead. The overhead of a cut gate depends on which gate is cut; see the final appendix of [`1 <https://arxiv.org/abs/2205.00016>`__] for details.  Here, we will focus on the CNOT gate.  If no real-time classical communication is available between qubits of the cut gate or wire, cut CNOT gates incur a sampling overhead of O(:math:`9^n`), and wire cuts incur a sampling overhead of O(16\ :sup:`n`), where n is the total number of cuts. If real-time communication is available (i.e., if the hardware supports “dynamic circuits”), the sampling overhead for both CNOT gate and wire cuts may be reduced to O(:math:`4^n`) [`1 <https://arxiv.org/abs/2205.00016>`__] [`3 <https://arxiv.org/abs/2302.03366>`__]; however, support for circuit cutting with classical communication (LOCC) is not yet supported in CKT.

Reviewer: Tweaking my own contribution :)

```suggestion
As mentioned above, the cost of any simulation based on quasiprobability distribution is an exponential sampling overhead. The overhead of a cut gate depends on which gate is cut; see the final appendix of [`1 <https://arxiv.org/abs/2205.00016>`__] for details.  Here, we will focus on the CNOT gate.  If no real-time classical communication is available between qubits of the cut gate or wire, cut CNOT gates incur a sampling overhead of O(:math:`9^n`), and wire cuts incur a sampling overhead of O(:math:`16^n`), where :math:`n` is the total number of cuts. If real-time communication is available (i.e., if the hardware supports “dynamic circuits”), the sampling overhead for both CNOT gate and wire cuts may be reduced to O(:math:`4^n`) [`1 <https://arxiv.org/abs/2205.00016>`__] [`3 <https://arxiv.org/abs/2302.03366>`__]; however, support for circuit cutting with classical communication (LOCC) is not yet supported in CKT.
```
